### PR TITLE
Add options to hide args and env vars

### LINKED ISSUE #2222: Feature request: option to hide environment variables
We'd like to make weave scope available to all of our developers.
Many of the pods have secrets injected as environment variables. We'd like to be able to either have the scope app not show these, or have the probe not report them in the first place.

### PR DESCRIPTION
**Pull request by mikebryant**
To allow for use of weave-scope in an unauthenticated environment,
add options to the probe to hide comand line arguments and
environment variables, which might contain secret data.

Fixes #2222

### CODE REVIEW

**Comment by 2opremio** [comment]
Thanks a lot for this! I will try to review it before of the day (Judging from a preliminary, quick look it seems to be good. You even included tests!).

File: probe/docker/registry.go [hunk 1]
```diff
@@ -94,7 +96,7 @@ func newDockerClient(endpoint string) (Client, error) {
 }
 
 // NewRegistry returns a usable Registry. Don't forget to Stop it.
-func NewRegistry(interval time.Duration, pipes controls.PipeClient, collectStats bool, hostID string, handlerRegistry *controls.HandlerRegistry, dockerEndpoint string) (Registry, error) {
+func NewRegistry(interval time.Duration, pipes controls.PipeClient, collectStats bool, hostID string, handlerRegistry *controls.HandlerRegistry, dockerEndpoint string, noCommandLineArguments bool, noEnvironmentVariables bool) (Registry, error) {
```
Comment (2opremio) [inline probe/docker/registry.go]: The argument list is too long (admittedly it was already pretty long). Mind moving them to a struct?

**Comment by 2opremio** [comment]
LGTM (modulo minor comment)

File: probe/docker/registry.go (hunk 1 above)
Comment (mikebryant) [inline probe/docker/registry.go]: err... Now we go into the fact that it's my first time writing in Golang.

Not entirely sure what that would look like - would you be able to point me to an example elsewhere in the code of what that looks like?

Comment (2opremio) [inline probe/docker/registry.go]: Just create something like

```
type RegistryOptions struct {
  Arg1 string
  Arg2 string
  ...
}
```

And pass it to `NewRegistry(options Registry options)`

Comment (mikebryant) [inline probe/docker/registry.go]: Updated, how does that look?

Comment (2opremio) [inline probe/docker/registry.go]: Much better, thanks. I will take another look tomorrow first thing.

File: probe/docker/controls_test.go [hunk 2]
```diff
@@ -59,7 +68,16 @@ func TestPipes(t *testing.T) {
 	mdc := newMockClient()
 	setupStubs(mdc, func() {
 		hr := controls.NewDefaultHandlerRegistry()
-		registry, _ := docker.NewRegistry(10*time.Second, nil, false, "", hr, "")
+		registry, _ := docker.NewRegistry(docker.RegistryOptions{
+			Interval:               10 * time.Second,
+			Pipes:                  nil,
```
Comment (2opremio) [inline probe/docker/controls_test.go]: There is no need to initialize fields getting the default value (nil, false and  "")

File: probe/docker/registry_test.go [hunk 3]
```diff
@@ -22,7 +22,16 @@ import (
 
 func testRegistry() docker.Registry {
 	hr := controls.NewDefaultHandlerRegistry()
-	registry, _ := docker.NewRegistry(10*time.Second, nil, true, "", hr, "")
+	registry, _ := docker.NewRegistry(docker.RegistryOptions{
+		Interval:               10 * time.Second,
+		Pipes:                  nil,
```
Comment (2opremio) [inline probe/docker/registry_test.go]: Same here

File: prog/probe.go [hunk 4]
```diff
@@ -195,7 +195,16 @@ func probeMain(flags probeFlags, targets []appclient.Target) {
 				log.Errorf("Docker: problem with bridge %s: %v", flags.dockerBridge, err)
 			}
 		}
-		if registry, err := docker.NewRegistry(flags.dockerInterval, clients, true, hostID, handlerRegistry, dockerEndpoint); err == nil {
+		if registry, err := docker.NewRegistry(docker.RegistryOptions{
```
Comment (2opremio) [inline prog/probe.go]: Optional: It's probably easier to read if you initialize the options separately

Comment (2opremio) [inline prog/probe.go]: (i.e. before the invocation of `NewRegistry()`)

**Comment by rade** [comment]
This has been reverted on master because it fails the lint check.

**Comment by 2opremio** [comment]
I merged it again at #2311 after fixing the linting problems ( linting didn't run in this PR due to #2192 )

**Comment by mikebryant** [comment]
Thanks. I'll remember to run those manually next time, sorry about that

**Pull request merged**